pkg/controller/projects/variables/controller.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package variables
+
+import (
+	"context"
+
+	"github.com/xanzy/go-gitlab"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane-contrib/provider-gitlab/apis/projects/v1alpha1"
+	"github.com/crossplane-contrib/provider-gitlab/pkg/clients"
+	"github.com/crossplane-contrib/provider-gitlab/pkg/clients/projects"
+)
+
+const (
+	errNotVariable  = "managed resource is not a Gitlab variable custom resource"
+	errGetFailed    = "cannot get Gitlab variable"
+	errCreateFailed = "cannot create Gitlab variable"
+	errUpdateFailed = "cannot update Gitlab variable"
+	errDeleteFailed = "cannot delete Gitlab variable"
+)
+
+// SetupVariable adds a controller that reconciles Variables.
+func SetupVariable(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha1.VariableKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.Variable{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.VariableGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newGitlabClientFn: projects.NewVariableClient}),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
```

Comment:
Nit: If you wanted, you could bump to use the latest `crossplane-runtime` and not pass in the `NewDefaultProviderConfig`. Building against the latest runtime will generate CRDs that use OpenAPI defaulting of `spec.providerConfigRef.name`, which saves us an `Update` call.

Reply:
@negz If not passing in the `NewDefaultProviderConfig`, ISTM that the `WithInitializers()` call is redundant too?